### PR TITLE
Fix line continuations in OBJ files

### PR DESCRIPTION
**Pull request by merwaaan**
Hi,
This fix to the OBJ parser reverts a faulty line from a January [commit](https://github.com/assimp/assimp/commit/35a044bda37b2bad50a41746622c746acd139631).

Some OBJ files have backslashes for line continuations, like in [this documentation](http://paulbourke.net/dataformats/obj/) or in OBJ files exported from Rhino for instance.

In such cases

`v 0.00000000001 0.00000000002 \`
`0.00000000003`

is equivalent to

`v 0.00000000001 0.00000000002 0.00000000003`

Assimp's OBJ parser dealt successfully with this case via the [getNextDataLine()](https://github.com/assimp/assimp/blob/475e51d157e7826bdcbc0d88e6a73b1764960e0a/include/assimp/IOStreamBuffer.h#L244) method which role is specifically to join lines separated by a continuation character (`\` in our case).

However, its usage was changed in the aforementioned commit. From [this issue](https://github.com/assimp/assimp/issues/2186) by the same author, it seems that they wanted to fix another bug related to material names and changed the continuation character but it actually broke OBJ parsing (the final data is wrong in most cases, and it crashes in other cases).

Tested on various obj files with continuations and now it works well, as before. The unit tests all pass.

### CODE REVIEW

**Comment by turol** [comment]
The build is broken but it appears unrelated to your changes. Strange...

Can you also add a testcase for this?

**Comment by merwaaan** [comment]
@turol, yeah I noticed that other PRs also fail some checks, something must be wrong.

I added a unit test, as requested.

File: test/unit/utObjImportExport.cpp [hunk 1]
```diff
@@ -448,3 +448,33 @@ TEST_F(utObjImportExport, import_without_linend) {
...
+    EXPECT_NEAR(vertices[0].z, 0.5f, threshold);
+
+    EXPECT_NEAR(vertices[1].x, -0.5f, threshold);
+    EXPECT_NEAR(vertices[1].y, -0.5f, threshold);
+    EXPECT_NEAR(vertices[1].z, -0.5f, threshold);
+
+    EXPECT_NEAR(vertices[2].x, -0.5f, threshold);
+    EXPECT_NEAR(vertices[2].y, 0.5f, threshold);
+    EXPECT_NEAR(vertices[2].z, -0.5f, threshold);
+}
```
Comment (turol) [inline test/unit/utObjImportExport.cpp]: Also add checks for number of vertices and primitives.

Comment (merwaaan) [reply test/unit/utObjImportExport.cpp]: Done.

**Comment by turol** [comment]
Looks good to me, now we just need someone to fix the build. @kimkulling ?

**Comment by kimkulling** [comment]
The broken build seems to get caused by a new compiler warning / error. Maybe they updated the clang / g++ version? I will prepare a fix for that!

**Comment by kimkulling** [comment]
Merged, thanks a lot for the contribution.

**Pull request merged**